velox/functions/sparksql/tests/DateTimeFunctionsTest.cpp

```diff
@@ -913,5 +913,61 @@ TEST_F(DateTimeFunctionsTest, fromUnixtime) {
       fromUnixTime(0, "yyyy-MM-dd HH:II"), "Specifier I is not supported");
 }
 
+TEST_F(DateTimeFunctionsTest, makeYMInterval) {
+  const auto fromYearAndMonth = [&](const std::optional<int32_t>& year,
+                                    const std::optional<std::int32_t>& month) {
+    auto result = evaluateOnce<int32_t, int32_t>(
+        "make_ym_interval(c0, c1)",
+        {year, month},
+        {INTEGER(), INTEGER()},
+        std::nullopt,
+        {INTERVAL_YEAR_MONTH()});
+    VELOX_CHECK(result.has_value());
+    return INTERVAL_YEAR_MONTH()->valueToString(result.value());
+  };
+  const auto fromYear = [&](const std::optional<int32_t>& year) {
+    auto result = evaluateOnce<int32_t, int32_t>(
+        "make_ym_interval(c0)",
+        {year},
+        {INTEGER()},
+        std::nullopt,
+        {INTERVAL_YEAR_MONTH()});
+    VELOX_CHECK(result.has_value());
+    return INTERVAL_YEAR_MONTH()->valueToString(result.value());
+  };
+
+  EXPECT_EQ(fromYearAndMonth(1, 2), "1-2");
+  EXPECT_EQ(fromYearAndMonth(0, 1), "0-1");
+  EXPECT_EQ(fromYearAndMonth(1, 100), "9-4");
+  EXPECT_EQ(fromYear(0), "0-0");
+  EXPECT_EQ(fromYear(178956970), "178956970-0");
+  EXPECT_EQ(fromYear(-178956970), "-178956970-0");
+  {
+    // Test signature for no year and month.
+    auto emptySignature = evaluateOnce<int32_t>(
```

Comment:
nit: emptySignature name is confusing since the value is not the signature; perhaps, rename to 'result' for zeroInterval.